src/workarounds/forwarddiff_rules.jl

```diff
@@ -180,15 +180,30 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
 
     ## Implicit differentiation
     hamψ_dual = ham_dual * ψ_dual
-    δHψ = [ForwardDiff.partials.(δHψₖ, 1) for δHψₖ in hamψ_dual]
-    δψ, response = solve_ΩplusK(basis, ψ, -δHψ, occupation;
-                                tol_cg=scfres.norm_Δρ, verbose=response.verbose)
-    δρ  = compute_δρ(basis, ψ, δψ, occupation)
+    δresults = ntuple(ForwardDiff.npartials(T)) do α
+        δHψα = [ForwardDiff.partials.(δHψk, α) for δHψk in hamψ_dual]
+        δψα, responseα = solve_ΩplusK(basis, ψ, -δHψα, occupation;
+                                      tol_cg=scfres.norm_Δρ, verbose=response.verbose)
+        δρα = compute_δρ(basis, ψ, δψα, occupation)
+        δψα, δρα, responseα
+    end
+    δψ = [δψα for (δψα, δρα, responseα) in δresults]
+    δρ = [δρα for (δψα, δρα, responseα) in δresults]
+    response = [responseα for (δψα, δρα, responseα) in δresults]
 
     ## Convert, combine and return
     DT = ForwardDiff.Dual{ForwardDiff.tagtype(T)}
-    ψ_out = [Complex.(DT.(real(ψₖ), real(δψₖ)), DT.(imag(ψₖ), imag(δψₖ))) for (ψₖ, δψₖ) in zip(ψ, δψ)]
-    ρ_out = DT.(scfres.ρ, δρ)
+    ψ_out = map(ψ, δψ...) do ψk, δψk...
+        map(ψk, δψk...) do ψi, δψi...
+            Complex(
+                DT(real(ψi), real.(δψi)),
+                DT(imag(ψi), imag.(δψi)),
+            )
```

Comment:
A bit less spaceous:
```suggestion
            Complex(DT(real(ψi), real.(δψi)),
                    DT(imag(ψi), imag.(δψi)))
```